### PR TITLE
evm/precompile: reject high s in erecover precompile

### PR DESCRIPTION
**Pull request by ptrus**
To make it consistent with with the Ethereum transaction signature verification (added in https://github.com/oasisprotocol/oasis-sdk/pull/452)

### CODE REVIEW

**Comment by ptrus** [comment]
Also what's with https://github.com/oasisprotocol/oasis-sdk/blob/44bff3160f517c94358935e80f7178a334b2ca36/runtime-sdk/modules/evm/src/precompile.rs#L45-L48

this is not used right?

**Comment by kostko** [comment]
> this is not used right?

I assume this "padded input" should be used from that point forward instead of "input" as otherwise this could panic if input is too small?

**Comment by ptrus** [comment]
> I assume this "padded input" should be used from that point forward instead of "input" as otherwise this could panic if input is too small?

Thought so as well, updated it as well. Thanks.

**Review by kostko — APPROVED** [review]

File: runtime-sdk/modules/evm/src/precompile.rs [hunk 1]
```diff
@@ -50,15 +50,15 @@ fn call_ecrecover(input: &[u8], target_gas: u64) -> Result<PrecompileOutput, Exi
     let mut sig = [0u8; 65];
 
     // input encoded as [hash, v, r, s]
-    msg[0..32].copy_from_slice(&input[0..32]);
-    sig[0..32].copy_from_slice(&input[64..96]);
-    sig[32..64].copy_from_slice(&input[96..128]);
+    msg[0..32].copy_from_slice(&padding[0..32]);
+    sig[0..32].copy_from_slice(&padding[64..96]);
+    sig[32..64].copy_from_slice(&padding[96..128]);
```
[the remark this answers is not in the document]
Comment (ptrus) [reply runtime-sdk/modules/evm/src/precompile.rs:55]: hm yeah i guess it separately copies `r` and `s`, no reason for it, will merge it into a single copy

**Pull request merged**